lib_spdif/src/spdif_chan_status.c

```diff
@@ -0,0 +1,227 @@
+// Copyright 2025 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+
+#include <stdint.h>
+#include <string.h>
+#include "spdif_pro_crc.h"
+
+uint8_t calc_spdif_pro_crc(uint32_t len, const uint8_t *buf, uint8_t init,
+                    uint8_t final_xor, uint8_t poly)
+{
+    uint8_t tmp = (init ^ buf[0]);
+    uint32_t xmos_crc = 0;
+    uint32_t shifted;
+    asm("crc8 %0, %1, %3, %4" : "=r"(xmos_crc), "=r"(shifted) : "0"(xmos_crc), "r"(tmp), "r"(poly));
```

Comment:
I think asm need to be marked volatile to ensure they dont get optimised out ever.

Reply:
BTW I think we should be using the following from xs1.h

`unsigned crc8shr(unsigned &checksum, unsigned data, unsigned poly);`

Reply:
oh. arent those only callable from xc? I could change this to an xc file I guess.